ccdproc/core.py

```diff
@@ -1575,6 +1576,163 @@ def cosmicray_median(ccd, error_image=None, thresh=5, mbox=11, gbox=0,
+    the column median box size should be at least 7 pixels. The median box can
+    be a single pixel wide along one dimension if needed. This may be
+    appropriate for spectroscopic long slit data.
+
+    The median subtracted image is then divided into blocks of size nclsig by
+    nlsig. In each block the pixel values are sorted and the pixels nearest the
+    30.9 and 69.1 percentile points are found; this would be the one sigma
+    points in a Gaussian noise distribution. The difference between the two
+    count levels divided by two is then the local sigma estimate. This
+    algorithm is used to avoid contamination by the bad pixel values. The block
+    size must be at least 10 pixels in each dimension to provide sufficient
+    pixels for a good estimate of the percentile sigma. The sigma uncertainty
+    estimate of each pixel in the image is then the sigma from the nearest
+    block.
+
+    The deviant pixels are found by comparing the median subtracted residual to
+    a specified sigma threshold factor times the local sigma above and below
+    zero (the lsigma and hsigma parameters). This is done for individual pixels
+    and then for column sums of pixels (excluding previously flagged bad
+    pixels) from two to the number of lines in the image. The sigma of the sums
+    is scaled by the square root of the number of pixels summed so that
+    statistically low or high column regions may be detected even though
+    individual pixels may not be statistically deviant. For the purpose of this
+    task one would normally select large sigma threshold factors such as six or
+    greater to detect only true bad pixels and not the extremes of the noise
+    distribution.
+
+    As a final step each column is examined to see if there are small segments
+    of unflagged pixels between bad pixels. If the length of a segment is less
+    than that given by the ngood parameter all the pixels in the segment are
+    also marked as bad.
+
+    Returns
+    -------
+    mask : `numpy.ndarray`
+        A boolean ndarray where the bad pixels have a value of 1 (True) and
+        valid pixels 0 (False), following the numpy.ma conventions.
+    """
+    try:
+        nlines, ncols = ratio.data.shape
+    except (TypeError, ValueError):
+        # shape is not iterable or has more or less than two values
+        raise ValueError('"ratio" must be two-dimensional.')
+    except AttributeError:
+        # No data attribute or data has no shape attribute.
+        raise ValueError('"ratio" should be a "CCDData".')
+
+    def _sigma_mask(baseline, one_sigma_value, lower_sigma, upper_sigma):
+        """Helper function to mask values outside of the specified sigma range.
+        """
+        return ((baseline < -lower_sigma * one_sigma_value) |
+                (baseline > upper_sigma * one_sigma_value))
+
+    mask = ~np.isfinite(ratio.data)
+    medsub = (ratio.data -
+              ndimage.median_filter(ratio.data, size=(nlmed, ncmed)))
```

Comment:
Yes there is. I haven't changed it because that's just a wrapper around `ndimage.median_filter` and at this line we're already working on the pure numpy arrays so there is no difference.

If you like it should be as easy as removing the `ndimage.median_filter` with `median_filter`.